traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -537,8 +542,12 @@ public HTTPRouteResult multiRoute(final HTTPRequest request, final Track track)
 			final List<Cache> caches = selectCaches(request, ds, track);
 
 			if (caches != null && !caches.isEmpty()) {
+				if (isClientSteeringDiversityEnabled() && caches.size() > selectedCaches.size()) {
+					caches.removeAll(selectedCaches);
+				}
 				final Cache cache = consistentHasher.selectHashable(caches, ds.getDispersion(), pathToHash);
 				steeringResult.setCache(cache);
+				selectedCaches.add(cache);
```

Comment:
Once all of the caches in the coverage zone have been added to the selectedCaches it looks like forced diversity becomes disabled and we revert to the standard consistent hash over all the caches. Is this the desired behavior? I would guess that you would want to clear the selectedCaches and start over.

Reply:
Yeah, that is the desired behavior that I wanted. We can't really assume that the set of caches assigned to each target is the same (even though in practice that might be true), otherwise I'd think we would do what you said and clear the selectedCaches to start over.

The effect this has is diversifying the result list "enough". E.g. with 3 available caches and 5 targets:
```
edge2.target1
edge1.target2
edge3.target3
<edge 1-3>.target4
<edge 1-3>.target5
```
The last two targets could be any of edges 1 through 3, but by that point we know the result already contains all the possible edges, so by the time we have failed over to target4, either all 3 edges were down, the first 3 target origins were down, or some mix of both. So we've at least covered our bases by trying all 3 possible caches before reaching targets 4 and 5 which could contain duplicates.